### ISSUE TITLE
uniqueItems property

### ISSUE THREAD

**Issue by DavidC0V** [YStrauch/swagger-object-validator#20]
Is it supported?
```
import * as SwaggerValidator from 'swagger-object-validator';
let validator = new SwaggerValidator.Handler();
 
let spec = {
    type: "array",
    uniqueItems: true,
    items: {
        type: "string",
        enum: ["1" ,"2", "3", "4"]
    }
}
 
let model = ['1', '2', '1']; // Should not be valid
validator.validateModel(model, spec, (err, result) => {
    console.log(result.humanReadable());
});
```

Oh.....found it:
```

if (schema.uniqueItems) {
--
37 | console.warn('WARNING: uniqueItems not supported (you found literally the only swagger feature that is currently unsupported)');


```

**Comment by DavidC0V** [comment]
Used patch `path-package` to do the following:
```
   if (schema.uniqueItems && test.length != [...new Set(test)].length) {
        pushError_1.pushError({
            errorType: result_1.ValidationErrorType.CONSTRAINTS_VIOATION,
            trace: trace,
            constraintName: 'uniqueItems',
            constraintValue: schema.uniqueItems
        }, errors, test, schema, spec, config);
     }
```
(Good enough for my purposes)

**Comment by YStrauch** [comment]
The problem with UniqueItems is, that swagger allows it to be placed on any array spec, even those containing very complex objects, in which case your Set implementation would fail. It would require iterating over all properties recursively and compare the two (or.. find a library to do that), which will hug even more performance. So I didn't implement it and added the warning instead.

If the feature is needed, I could implement it. Pull requests welcome and easy as it's an entirely separated check exactly on the line where the warning is generated. There's even a test in place with one level of complexity but it's commented out.

**Comment by DavidC0V** [comment]
I'm aware of that but the Set solution is good enough for my purposes (which are similar to the example).
If the performance impact is big you could always add it as part of the IValidatorConfig - with default false - and users can decide for themselves.

**Comment by YStrauch** [comment]
Hi!

I've implemented it within version 1.4.0

Please check the docs relating the default cap of max items and other potentially breaking changes in this version.
https://github.com/YStrauch/swagger-object-validator#limiting-the-unique-items-constraint
https://github.com/YStrauch/swagger-object-validator#140
